sqlmesh/core/snapshot/definition.py

```diff
@@ -252,8 +254,7 @@ class SnapshotTableInfo(PydanticModel, SnapshotInfoMixin, frozen=True):
     parents: t.Tuple[SnapshotId, ...]
     previous_versions: t.Tuple[SnapshotDataVersion, ...] = ()
     change_category: t.Optional[SnapshotChangeCategory]
-    is_materialized: bool
-    is_embedded_kind: bool
+    kind_name: ModelKindName
```

Comment:
why can't we just call this `model_kind_name` here and eliminate a need for the property method in this class?

Reply:
because then you can't override it, pydantic complains